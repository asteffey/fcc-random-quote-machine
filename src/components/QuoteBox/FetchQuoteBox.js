import React from 'react';
import { bindActionCreators } from 'redux';
import { retrieveQuote } from '../../actions';
import { connect } from 'react-redux';
import QuoteBox from './QuoteBox';

const FetchQuoteBox = ({id, retrieveQuote}) => {
    React.useEffect(() => {
        retrieveQuote(id);
    }, [id, retrieveQuote]);

    return (
        <QuoteBox />
    );
};

const mapStateToProps = state => ({});
const mapDispatchToProps = dispatch => bindActionCreators({ retrieveQuote }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(FetchQuoteBox);