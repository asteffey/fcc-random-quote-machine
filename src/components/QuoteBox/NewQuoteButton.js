import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const NewQuoteButton = ({ nextId }) => {

    return (
        <Link id="new-quote" to={`/${nextId}`}>New Quote</Link>
    );
};

NewQuoteButton.propTypes = {
    nextId: PropTypes.number.isRequired
};

export default NewQuoteButton;