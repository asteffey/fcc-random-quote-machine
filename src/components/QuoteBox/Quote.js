import React from 'react';
import PropTypes from 'prop-types';
import Skeleton from 'react-loading-skeleton';

const wordLengths = [3, 2, 8, 4, 8, 4, 3, 1, 4, 7, 1, 6, 6, 5, 4, 6, 9, 2, 3, 7, 4, 2, 7, 3, 3, 4, 4, 6, 2, 1, 5, 2, 5, 8, 3, 6, 9, 4, 3, 1, 6, 2, 9, 8, 11, 6];
const skeletonParagraph = wordLengths.flatMap((w, i) => [<Skeleton key={i} width={w * 9} />, ' ']);
const SkeletonText = () => (
    <span>{skeletonParagraph}</span>
);

const Quote = ({ text, author }) => {
    return (
        <div id="quote">
            <p id="text">{text || <SkeletonText />}</p>
            <p id="author">{author || <Skeleton width={200}/>}</p>
        </div>
    );
};

Quote.propTypes = {
    text: PropTypes.string,
    author: PropTypes.string
};

export default Quote;