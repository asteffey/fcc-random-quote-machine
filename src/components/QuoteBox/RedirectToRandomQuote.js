import React from 'react';
import { useSelector } from 'react-redux';
import { Redirect } from 'react-router-dom';

export default () => {
    const quoteId = useSelector(state => state.nextId);
    React.useEffect(()=>{
        console.log(`Redirecting to ${quoteId}`);
    });
    
    return (
        <Redirect to={`/${quoteId}`} />
    );
};