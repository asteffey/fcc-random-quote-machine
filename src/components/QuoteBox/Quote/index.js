import React from 'react';
import { useSelector } from 'react-redux';
import Quote from './Quote';
import Loading from '../../Loading';

export default () => {
    const status = useSelector(state => state.status);
    const quote = useSelector(state => state.quote);

    return (
        <Loading {...status}>
            <Quote {...quote} />
        </Loading>
    );

};