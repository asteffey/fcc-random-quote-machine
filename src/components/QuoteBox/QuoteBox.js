import React from 'react';
import Quote from './Quote';
import NewQuoteButton from './NewQuoteButton';
import TweetButton from '../TweetButton';
import useFetchQuote from '../../hooks/useFetchQuote';

export default ({ match: { params: { id } } }) => {
    useFetchQuote(id);

    return (
        <div id="quote-box">
            <Quote />
            <footer>
                <NewQuoteButton />
                <TweetButton />
            </footer>
        </div>
    );
};