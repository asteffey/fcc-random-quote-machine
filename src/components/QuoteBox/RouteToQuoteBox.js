import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import RedirectToRandomQuote from './RedirectToRandomQuote';
import QuoteBox from './QuoteBox';

export default () => (
    <Router>
        <Switch>
            <Route exact path="/" component={RedirectToRandomQuote} />
            <Route path="/:id" component={QuoteBox} />
        </Switch>
    </Router>
);