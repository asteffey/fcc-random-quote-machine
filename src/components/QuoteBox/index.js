import React from 'react';
import RouteToQuoteBox from './RouteToQuoteBox';

const QuoteBoxWrapper = () => (
    <RouteToQuoteBox/>
);

export default QuoteBoxWrapper;