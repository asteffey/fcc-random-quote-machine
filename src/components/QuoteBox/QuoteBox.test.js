import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import QuoteBox from './QuoteBox';

import useFetchQuote from '../../hooks/useFetchQuote';

jest.mock('./Quote', () => () => <div id="Quote"/>);
jest.mock('./NewQuoteButton', () => () => <div id="NewQuoteButton"/>);
jest.mock('../TweetButton', () => () => <div id="TweetButton"/>);
jest.mock('../../hooks/useFetchQuote');

describe('QuoteBox', () => {
    const testId = 1234;

    afterEach(() => {
        useFetchQuote.mockClear();
    })
  
    it('matches snapshot', () => {
        const { container } = render(<QuoteBox id={testId} />);
        expect(container).toMatchSnapshot();
    });

    it('creates #quote-box element', () => {
        const { container } = render(<QuoteBox id={testId} />);
        const quoteBox = container.querySelector('#quote-box');
        expect(quoteBox).toBeInTheDocument();
    });

    it('contains Quote', () => {
        const { container } = render(<QuoteBox id={testId} />);
        const Quote = container.querySelector('#Quote');
        expect(Quote).toBeInTheDocument();
    });

    it('contains NewQuoteButton', () => {
        const { container } = render(<QuoteBox id={testId} />);
        const NewQuoteButton = container.querySelector('#NewQuoteButton');
        expect(NewQuoteButton).toBeInTheDocument();
    });

    it('contains TweetButton', () => {
        const { container } = render(<QuoteBox id={testId} />);
        const TweetButton = container.querySelector('#TweetButton');
        expect(TweetButton).toBeInTheDocument();
    });

    it('fetches quote', () => {
        render(<QuoteBox id={testId} />);
        expect(useFetchQuote.mock.calls.length).toBe(1);
        expect(useFetchQuote.mock.calls[0][0]).toBe(testId);
    });


});