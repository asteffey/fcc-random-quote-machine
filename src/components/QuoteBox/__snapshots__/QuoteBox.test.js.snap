// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteBox matches snapshot 1`] = `
<div>
  <div
    id="quote-box"
  >
    <div
      id="Quote"
    />
    <footer>
      <div
        id="NewQuoteButton"
      />
      <div
        id="TweetButton"
      />
    </footer>
  </div>
</div>
`;
